packages/aws-serverless/src/utils.ts

```diff
@@ -12,3 +36,47 @@ export function markEventUnhandled(scope: Scope): Scope {
 
   return scope;
 }
+
+/**
+ * Extracts sentry trace data from the handler `context` if available and falls
+ * back to the `event`.
+ */
+export function getTraceData(event: HandlerEvent, context?: HandlerContext): TraceData {
+  const headers = event.headers || {};
+
+  const traceData: TraceData = {
+    'sentry-trace': headers['sentry-trace'],
+    baggage: headers.baggage,
+  };
+
+  if (context && context.clientContext && context.clientContext.Custom) {
+    const customContext: Record<string, unknown> = context.clientContext.Custom;
+    const sentryTrace = isString(customContext['sentry-trace']) ? customContext['sentry-trace'] : undefined;
+
+    if (sentryTrace) {
```

Comment:
m: It will probably not matter in 99% of cases, but let's just split this into two if-blocks. So basically:

```js
const sentryTrace = ...;
const baggage = ...;

if(sentryTrace) {
  traceData['sentry-trace'] = sentryTrace;
}
if(baggage) {
  traceData.baggage = baggage;
}
```

I think this is slightly more robust :)

Reply:
I was thinking of that, but wouldn't we potentially end up with a mix of wrong pairs, e.g. both `event` and `context` have `baggage` but only `event` has `sentry-trace`. So we end up with `sentry-trace` from `event` and `baggage` from `context`.

I don't know if that's a realistic usecase tho. In core, we  also throw away `baggage` if `sentryTrace` is not valid.

Reply:
right, then let's keep it this way, all good! 👍